### PR TITLE
crossplane: add support for attribute-based APIs

### PR DESCRIPTION
**Pull request by muvaf**
Description of changes: Adds support for attribute-based APIs such as SNS to Crossplane pipeline. All ACK tests pass, there is no change in output of ACK pipeline.

/cc @jaypipes 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/crossplane/boilerplate.go.tpl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{- define "boilerplate" -}}
 /*
-Copyright 2020 The Crossplane Authors.
+Copyright 2021 The Crossplane Authors.
```
Comment (jaypipes) [inline templates/crossplane/boilerplate.go.tpl:3]: This is why I always leave dates out of headers like this :)

File: templates/crossplane/pkg/controller.go.tpl [hunk 2]
```diff
@@ -159,8 +171,14 @@ func newExternal(kube client.Client, client svcsdkapi.{{ .SDKAPIInterfaceTypeNam
...
 		kube:           kube,
 		client:         client,
+		{{- if or .CRD.Ops.ReadOne .CRD.Ops.GetAttributes .CRD.Ops.ReadMany }}
 		preObserve:     nopPreObserve,
 		postObserve:    nopPostObserve,
+		lateInitialize: nopLateInitialize,
+    isUpToDate:     alwaysUpToDate,
+    {{- else }}
+    observe:        nopObserve,
+    {{- end }}
```
Comment (jaypipes) [inline templates/crossplane/pkg/controller.go.tpl:181]: looks like indentation is goofy above?

**Review by jaypipes — APPROVED** [review]
:+1: 

Nice stuff @muvaf!

File: templates/crossplane/pkg/controller.go.tpl (hunk 2 above)
Comment (muvaf) [reply templates/crossplane/pkg/controller.go.tpl]: Went over all indentations and fixed them.

**Review by jaypipes — APPROVED** [review]
excellent, thanks for fixing those indentations!

**Pull request merged**